src/compiler/scala/tools/nsc/transform/Erasure.scala

```diff
@@ -1226,13 +1226,19 @@ abstract class Erasure extends InfoTransform
         case Match(selector, cases) =>
           Match(Typed(selector, TypeTree(selector.tpe)), cases)
 
-        case Literal(ct) if ct.tag == ClazzTag
-                         && ct.typeValue.typeSymbol != definitions.UnitClass =>
-          val erased = ct.typeValue.dealiasWiden match {
-            case tr @ TypeRef(_, clazz, _) if clazz.isDerivedValueClass => scalaErasure.eraseNormalClassRef(tr)
-            case tpe => specialScalaErasure(tpe)
+        case Literal(ct) =>
+          val cleanLiteral = {
+            if (tree.attachments.get[OriginalTreeAttachment].isEmpty) tree
```

Comment:
Use `contains` instead?

Reply:
Done.

Reply:
Or better yet, just use `remove` directly. If this pattern of checking whether the attachment is there before removing is faster, we should implement it once in `remove` (one method call that requires a class tag is also better than two)

Reply:
Done in last commit!